app/models/hyrax/file_set.rb

```diff
@@ -73,6 +73,15 @@ def original_file_id
       original_file&.id
     end
 
+    # @return [String, Nil] versioned identifier suitable for use in a IIIF manifest
+    def iiif_id
+      orig_file = original_file
+      return nil if orig_file.nil? || orig_file.file_identifier.blank?
+      latest_file = Hyrax::VersioningService.latest_version_of(orig_file)
+      version = latest_file&.version_id ? Digest::MD5.hexdigest(latest_file.version_id) : nil
+      "#{id}/files/#{orig_file.id}#{'/' + version if version}"
```

Comment:
If version is `nil`, this would still return a string missing the version number. Is this the expected behavior?

Reply:
Would `"#{id}/files/#{orig_file.id}"` still be a valid `iiif_id` when `version` is `nil`, or would it be more accurate for the method to return `nil`?

Reply:
I consider it valid. A IIIF server that is resolving requests for versionless ids would not know if its cached image times have become outdated due to a fileset revision, but that seems preferable to not working at all if an unversioned storage adapter is used.